packages/commonwealth/client/scripts/views/components/share_popover.tsx

```diff
@@ -72,14 +78,19 @@ export const SharePopover = ({
           iconLeftSize: 'regular',
           label: 'Share on X (Twitter)',
           onClick: async () => {
-            if (!commentId) {
-              await window.open(
-                `https://twitter.com/intent/tweet?text=${domain}${discussionLink}`,
+            if (discussionLink) {
+              window.open(
+                `${twitterPrefix}${domain}${discussionLink}`,
+                '_blank',
+              );
+            } else if (commentId) {
+              window.open(
```

Comment:
Share on twitter seems to be broken

Uploading Screen Recording 2024-04-26 at 7.06.14 PM.mov…